python_modules/dagster/dagster/_core/run_coordinator/queued_run_coordinator.py

```diff
@@ -242,6 +268,45 @@ def submit_run(self, context: SubmitRunContext) -> DagsterRun:
             check.failed(f"Failed to reload run {dagster_run.run_id}")
         return run
 
+    def _attach_op_concurrency_info(self, context: SubmitRunContext) -> None:
+        run = context.dagster_run
+        if not run.external_job_origin:
+            return
+        workspace = context.workspace
+        location = workspace.get_code_location(run.external_job_origin.location_name)
+        subset_result = location.get_subset_external_job_result(
+            JobSubsetSelector(
+                location_name=run.external_job_origin.location_name,
+                repository_name=run.external_job_origin.external_repository_origin.repository_name,
+                job_name=run.external_job_origin.job_name,
+                op_selection=None,
+                asset_selection=run.asset_selection,
+            )
+        )
```

Comment:
Needing to do an additional gRPC round-trip here at submit time is not ideal when this information was almost certainly already fetched back when this run was created a few seconds ago - is there a way to fetch this information at that stage somehow? Recognize that that is a bigger change

Reply:
Yeah, going to have to punt on this for now...

The changes I made make sure to add this information only when the plan snapshot is already in scope (e.g. at run creation time) to avoid the extra fetches, but we probably could figure out all this info from the full job snapshot plus asset selection.  Would need to port that subselection code over to operate w/o the location.